### DISCUSSION TITLE
otUDPnewMessage unable to allocate memory at faster rate

### DISCUSSION THREAD

**Discussion by nikhil1983** [openthread/openthread#8272] [Q&A] [answered]
Hello, 

I am using the Nordic mqtt-sn example as the base for developing my thread solution. 

So my Sleepy end device has two types of data collection modes: 

1. Data sent every 20 sec in the form of a bunch of payload consisting of sensor samples read at every 100 msec. This works fine.

2. Data sent every 100 milli sec using Mqtt-sn publish function which call otudpnewmessage. But after a few seconds, the function gives an error of Failed to allocate memory. Also at this point of time I only have a single node in the network transmitting at such a fast rate 100ms of other nodes are sending data at 1 minute intervals. 

Also in my example I haven't used any dynamic memory allocation function such as malloc and nrf_malloc (Nordic) etc. 

So I would like to know why my single node in the network is unable to publish at the rate of 100ms interval continuously . 
Also I would like to get few suggestions for the same because I want to plot the data samples of sensor continuously on graph. 

So I need continuous data instead of buffered data which will make graph look lagging. 

Hoping for a quick response.

**Comment by jwhui** [comment, marked as answer]
OpenThread maintains a message buffer pool for allocating messages (including UDP messages). The size of the message buffer pool is specified using the following build option: https://github.com/openthread/openthread/blob/9ebe41f16e5107060abbdeb3a7ba26f1d7373b1d/src/core/config/misc.h#L170

If the offered load is less than what Thread can deliver, then you can try increasing the message buffer pool size. However, if the offered load is higher than what Thread can deliver, then you will need to rethink how many messages your application generates.
